### ISSUE TITLE
Report missed `functools.wraps` for decorators

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1707]
# Rule request

## Thesis

```python
# bad
def a(f):
    ...
    def b():
        ...
    return b

import functools

# good
def a(f):
    ...
    @functools.wraps
    def b():
        ...
    return b

# good
def a(f):
    ...
    def b():
        ...
    b = functools.update_wrapper(b, f)
    return b

# good
def a(f):
    ...
    def b():
        ...
    functools.update_wrapper(b, f)
    return b
```

## Reasoning

Decorators should always preserve the original function for the sake of introspection.

**Comment by sobolevn** [comment]
Let's leave this for `typed-linter`. It is going to be hard without types.

**Comment by orsinium** [comment]
Another idea for a rule: catch wrong arguments order for `update_wrapper`. I don't think it can be done by MyPy since both arguments are just `Callable`.

**Comment by sobolevn** [comment]
I am using `typed-linter` for all rules that we would need to rebase there.

**Comment by orsinium** [comment]
@sobolevn should we add `typed-linter` tag?

**Comment by sobolevn** [comment]
Yes, good idea!
